src/TextShow/Data/VectorNonEmpty.hs

```diff
@@ -0,0 +1,27 @@
+{-# LANGUAGE TemplateHaskell #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
+{-|
+Module:      TextShow.Data.Vector.NonEmpty
+Copyright:   (C) 2014-2023 Ryan Scott
+License:     BSD-style (see the file LICENSE)
+Maintainer:  Ryan Scott
+Stability:   Provisional
+Portability: GHC
+
+'TextShow' instances for @NonEmptyVector@ types.
+-}
+module TextShow.Data.VectorNonEmpty () where
```

Comment:
For consistency with how `nonempty-vector` names its modules, let's name this module `TextShow.Data.Vector.NonEmpty` instead.